colcon_cmake/task/cmake/__init__.py

```diff
@@ -208,3 +210,72 @@ def get_visual_studio_version():
     :rtype: str
     """
     return os.environ.get('VisualStudioVersion', None)
+
+
+def _parse_cmake_version():
+    """
+    Parse the CMake version number by running `cmake --version`.
+
+    The version number is parse from the output of 'cmake --version' and is
+    expected in the form 'cmake version #.#.#' where each '#' character
+    represents part of the version number. Additional text may follow, however
+    only the first line is parsed. Parsing is performed by the pkg_resources
+    package and the returned object is a Version object from that package.
+
+    This function blocks on the execution of cmake and should only be used to
+    cache the CMake version number. External API users should use
+    get_cmake_version()
+
+    :returns: The version as parsed by the pkg_resources package
+    :rtype pkg_resources.extern.packaging.version.Version
+    """
+    try:
+        output = subprocess.check_output([CMAKE_EXECUTABLE, '--version'])
+        lines = output.decode().splitlines()
+        ver_line = lines[0] if lines and len(lines) else None
+        if ver_line:
+            # Extract just the version part of the string.
+            ver_re_str = r'^(?:.*)(\d+\.\d+\..*)'
+            ver_match = re.match(ver_re_str, ver_line, re.I)
+            if ver_match:
+                return parse_version(ver_match.group(1))
+        return None
+    except subprocess.CalledProcessError:
+        return None
+    return None
+
+
+# Global variable for the cached CMake version number.
+# When valid, this will be of pkg_resources.extern.packaging.version.Version
+# It may also have a boolean value False when the cmake version has failed
+# to parse. This saves recurring parse failures.
+_cached_cmake_version = None
+
+
+def get_cmake_version():
+    """
+    Get the cached CMake version number if available and successfully parsed.
+
+    The result is None when a version number is not available or cannot be
+    parsed. This may occur if CMake is not present, is a very old version or is
+    a newer version where the version number string output has changed.
```

Comment:
For what never version is that the case? Then the `--install` option wouldn't be used.

Reply:
I'm identifying the behaviour should the cmake version string format change in future. More of a maintenance note.

Reply:
I've modified the comment slightly.